mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,709 @@
+        ...
+        num_proposals
+        label, best_iou, overlap_self, start_frame, end_frame
+        label, best_iou, overlap_self, start_frame, end_frame
+        ...
+    Example of a standard annotation file:
+    .. code-block:: txt
+        # 0
+        video_validation_0000202
+        5666
+        1
+        3
+        8 130 185
+        8 832 1136
+        8 1303 1381
+        5
+        8 0.0620 0.0620 790 5671
+        8 0.1656 0.1656 790 2619
+        8 0.0833 0.0833 3945 5671
+        8 0.0960 0.0960 4173 5671
+        8 0.0614 0.0614 3327 5671
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        data_prefix (str): Path to a directory where videos are held.
+            Default: None.
+        test_mode (bool): Store True when building test or validation dataset.
+            Default: False.
+        filename_tmpl (str): Template for each filename.
+            Default: 'img_{:05}.jpg'.
+        video_centric (bool): Whether to sample proposals just from
+            this video or sample proposals randomly from the entire dataset.
+            Default: True.
+        reg_stats (list): Regression target normlizing constants, including
+            mean and standard deviation of location and duration.
+        body_segments (int): Number of segments in course period.
+            Default: 5.
+        aug_segments (list[int]): Number of segments in starting and
+            ending period. Default: (2, 2).
+        aug_ratio (int | float | tuple[int | float]): The ratio of the length
+            of augmentation to that of the proposal. Defualt: (0.5, 0.5).
+        clip_len (int): Frames of each sampled output clip.
+            Default: 1.
+        frame_interval (int): Temporal interval of adjacent sampled frames.
+            Default: 1.
+        filter_gt (bool): Whether to filter videos with no annotation
+            during training. Default: True.
+        verbose (bool): Whether to print full information or not.
+            Default: False.
+    """
+
+    def __init__(self,
+                 ann_file,
+                 pipeline,
+                 train_cfg,
+                 test_cfg,
+                 data_prefix=None,
+                 test_mode=False,
+                 filename_tmpl='img_{:05d}.jpg',
+                 video_centric=True,
+                 reg_stats=None,
+                 body_segments=5,
+                 aug_segments=(2, 2),
+                 aug_ratio=(0.5, 0.5),
+                 clip_len=1,
+                 frame_interval=1,
+                 filter_gt=True,
+                 verbose=False):
+        super(SSNDataset, self).__init__(ann_file, pipeline, data_prefix,
+                                         test_mode)
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self.assigner = train_cfg.ssn.assigner
+        self.sampler = train_cfg.ssn.sampler
```

Comment:
Can we directly pass a `assigner` and `sampler` dict to it? It seems `train_cfg` and `test_cfg` are unused.

Reply:
But other parts(like model, losses) will use parameters of `assigner` and `sampler` too, so we can fetch those parameters from `train_cfg` and `test_cfg` in all components of SSN.

Reply:
OK, let's try it first